src/test/java/com/browserstack/examples/rule/WebDriverProviderRule.java

```diff
@@ -29,9 +40,44 @@ public String getMethodName() {
         return methodName;
     }
 
-    public WebDriver getWebDriver(WebDriverConfiguration webDriverConfiguration, Platform platform) {
+    public WebDriver getWebDriver(WebDriverConfiguration webDriverConfiguration, Platform platform) throws MalformedURLException {
         this.webDriverConfiguration = webDriverConfiguration;
-        // instantiate the WebDriver based on whether it is a remote or a local driver
+        switch (webDriverConfiguration.getDriverType()) {
+            case localDriver:
+                System.setProperty(WEBDRIVER_CHROME_DRIVER, Paths.get(platform.getDriverPath()).toString());
+                this.driver = new ChromeDriver();
+                break;
+            case remoteDriver:
+                RemoteDriverConfig remoteDriverConfig = webDriverConfiguration.getRemoteDriver();
```

Comment:
Rename this method to getRemoteDriverConfig

Reply:
Done